src/main/java/wolox/training/models/Users.java

```diff
@@ -0,0 +1,85 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Vector;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+public class Users {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long id;
+
+  @Column(nullable = false)
+  private String username;
+
+  @Column(nullable = false)
+  private String name;
+
+  @Column(nullable = false)
+  private LocalDate birthday;
+
+  @ManyToMany(cascade = CascadeType.ALL)
```

Comment:
This CascadeType is wrong.
By saying ALL, you are saying: Persist, Merge, Refresh, Delete and Detach.

Detach is only so that something get's detached along the cascade when you detach an object from the persistence context manually. This almost never happens, I've actually never seen it in my life.

Persist says: If you persist the Users instance to the DB, then persist every item that this collection has in the DB as well.
Delete says: If you delete the Users instance from the DB, then delete every item that this collection has in the DB as well.

Merge says: If you changed the in-memory values of the collection, then merge this changes to the DB so these are reflected there.
Refresh says: If something changed in the database values of the collection, then refresh those changes in the in-memory values.

Given a Book can be created and deleted independantly of it being associated to a specific user, you don't wan't to cascaded persist nor delete in this case.

The only thing you want, is to keep the JoinTable updated.
That's only Refresh + Merge.
:)